bayes_opt/domain_reduction.py

```diff
@@ -97,15 +97,23 @@ def _update(self, target_space: TargetSpace) -> None:
         self.r = self.contraction_rate * self.r
 
     def _trim(self, new_bounds: np.array, global_bounds: np.array) -> np.array:
-        for i, variable in enumerate(new_bounds):
-            if variable[0] < global_bounds[i, 0]:
-                variable[0] = global_bounds[i, 0]
-            if variable[1] > global_bounds[i, 1]:
-                variable[1] = global_bounds[i, 1]
+        """ Tests new_bounds to enforce global_bounds and minimum_window."""
+        # sort the order of the bounds for each parameter
+        for i, pbounds in enumerate(new_bounds):
+            if pbounds[0] > pbounds[1]:
+                new_bounds[i, 0] = pbounds[1]
+                new_bounds[i, 1] = pbounds[0]
+
+        # check if the new bounds exceed the global bounds
+        for i, pbounds in enumerate(new_bounds):
+            # check if lower bound exceeds range 
+            if (pbounds[0] < global_bounds[i, 0] or pbounds[0] > global_bounds[i, 1]):
+                pbounds[0] = global_bounds[i, 0]
+            # check if upper bound exceeds range
+            if (pbounds[1] > global_bounds[i, 1] or pbounds[1] < global_bounds[i, 0]):
+                pbounds[1] = global_bounds[i, 1]
```

Comment:
This can essentially cause a reset to the initial global bounds if we pan outside them completely.
1. Can this case occur naturally, or only if the user manually messes with the bounds?
2. If it can happen naturally, can the bounds transformer recover from this and resume normal operation?

Reply:
FYI, I think this possible route to non-convergence might already be present in `v1.4.3`